### DISCUSSION TITLE
Proposal to replace fabric-sdk-go

### DISCUSSION THREAD

**Discussion by 1gezhanghao** [hyperledger/fabric-gateway#500] [Q&A]
As we know fabric-sdk-go （https://github.com/hyperledger/fabric-sdk-go） has not been updated for a long time. It has too many code conflicts with fabric v2.4+
Since the new Fabric-gateway API been released, It already has the ability to interact with system chaincode,
which means all  the features of fabric-sdk-go can be implemented in the fabric-gateway project
So, is there an official plan to replace fabric-sdk-go

**Comment by bestbeforetoday** [comment]
The current plan is for all the legacy SDKs ([fabric-sdk-go](https://github.com/hyperledger/fabric-sdk-go), [fabric-sdk-node](https://github.com/hyperledger/fabric-sdk-node), [fabric-gateway-java](https://github.com/hyperledger/fabric-gateway-java), [fabric-sdk-java](https://github.com/hyperledger/fabric-sdk-java)) to be deprecated once Fabric v2.5 is released as a long-term support (LTS) release.

The Fabric Gateway client API is intended to support the development of business applications with capability to submit and evaluate transactions, and to receive events. This matches the capabilities provided by the Fabric programming model in the Go SDKs [gateway](https://pkg.go.dev/github.com/hyperledger/fabric-sdk-go/pkg/gateway) package, Node SDKs [fabric-network](https://hyperledger.github.io/fabric-sdk-node/release-2.2/module-fabric-network.html) module, and Java [fabric-gateway-java](https://hyperledger.github.io/fabric-gateway-java/) package. There is no plan to extend the Fabric Gateway client API to provide the admin capability that exists in some of the lower-level legacy SDKs. As you say, many of those capabilities can be implemented by using this API to invoke transactions functions provided by system chaincode.

**Comment by 1gezhanghao** [comment]
As all the legacy SDKs will be deprecated in Fabric v2.5. People still need admin capabilities, no matter implements in fabric-gateway or start a new sdk project. It's too complicated for people to implments a lower-level SDK direct use system chaincode  by  themsleves, there are too many details to deal. 

There is a new gateway mechanism in the fabric project that fabric-gateway API will connect to. I am not sure the peer cmd use it or not.
but if they use or will use the  same way, extracting a new admin capabilities sdk from it will be easy and necessary . It's better to release a new low-lever-sdk with fabric V2.5+ by the official.

**Reply by bestbeforetoday** [reply]
For some time now the recommendation has been to use the CLI commands for admin operations. This is partly because the legacy SDKs don't provide a very consistent set of capability, partly because there was a deliberate focus of the newer SDKs on supporting business application runtime (not admin operations), and partly because there was not enough community support to implement and maintain admin APIs in all the client languages. I appreciate that for some users this is not the ideal solution.

The current Fabric admin CLI commands do not use the new Gateway service. Some of the existing Gateway capability could be used to simplify the implementation of admin APIs. There are also some areas where the Gateway service is not such a good fit. For example, some admin operations require targeting specific network peers. Gateway was specifically designed to avoid the need for client applications to know about or target specific peers, since having to know implementation details and topology of a deployed Fabric network is not helpful when implementing business applications.

A group of contributors are working on a new admin SDK here, which I think is a great initiative: https://github.com/Hyperledger-TWGC/fabric-admin-sdk
